core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -1096,6 +1096,11 @@ private[spark] class TaskSetManager(
   def executorAdded() {
     recomputeLocality()
   }
+
+  def markPartitionIdAsCompletedForTaskAttempt(index: Int): Unit = {
```

Comment:
Also, I think we need to increase `tasksSuccessful` and mark this `TaskSetManager` as zombie if  its successful tasks num has reached `numTasks`. You can refer to `markPartitionCompleted()` for details.

Reply:
Have replied to your question in the below comment. Let me know your thoughts. Thank you.